### PR TITLE
Create Marker on Middle

### PR DESCRIPTION
**Pull request by julialoeschel**
create middle (line distance) between points and set marker 

Bildschirmfoto 2021-12-01 um 18 24 33

#13

### CODE REVIEW

File: src/app/components/Map/Map.tsx [hunk 1]
```diff
@@ -212,7 +214,26 @@ export default function MapBox(): JSX.Element {
...
+    const point2 = location2 ? turf.point(location2) : null
+    const point1Coords = point1?.geometry.coordinates
+    const point2Coords = point2?.geometry.coordinates
+
+    const midpoint = turf.midpoint(
+      point1Coords as GeoJSON.Position,
+      point2Coords as GeoJSON.Position
+    )
+    const midpointCoords = midpoint.geometry.coordinates
+    console.log(midpointCoords)
```
Comment (PReichetanz) [inline src/app/components/Map/Map.tsx:228]: You can delete this now :)

**Review by PReichetanz — APPROVED** [review]
Looks good to me 😊

File: package.json [hunk 2]
```diff
@@ -49,6 +49,10 @@
   "dependencies": {
     "@mapbox/mapbox-gl-directions": "^4.1.0",
     "@mapbox/mapbox-gl-geocoder": "^4.7.4",
+    "@turf/buffer": "^6.5.0",
+    "@turf/collect": "^6.5.0",
+    "@turf/midpoint": "^6.5.0",
+    "@turf/turf": "^6.5.0",
```
Comment (marwinburesch) [inline package.json:55]: So these are all for getting the middle of the route? I presume you need all of them. However, you forgot the [types](https://www.npmjs.com/package/@types/turf) :) (devDependencies)

File: src/app/components/Map/Map.tsx [hunk 3]
```diff
@@ -212,7 +214,26 @@ export default function MapBox(): JSX.Element {
...
     }
+    // get middle point between location1 and location2 (strait line)
+    const point1 = location1 ? turf.point(location1) : null
+    const point2 = location2 ? turf.point(location2) : null
+    const point1Coords = point1?.geometry.coordinates
+    const point2Coords = point2?.geometry.coordinates
+
+    const midpoint = turf.midpoint(
+      point1Coords as GeoJSON.Position,
+      point2Coords as GeoJSON.Position
```
Comment (marwinburesch) [inline src/app/components/Map/Map.tsx:225]: This will likely be obsolete after getting the types package

**Review by marwinburesch — COMMENTED** [review]
nice

File: package.json (hunk 2 above)
Comment (julialoeschel) [reply package.json:55]: no, all is included in turf/turf (also types). I deleted the rest

Comment (marwinburesch) [reply package.json:55]: Thanks, I just saw the package is deprecated
cheers!

**Pull request merged**